core/language.go

```diff
@@ -12,6 +12,11 @@ type LanguageResolvers interface {
 	// ResolveImports returns a list of import statements
 	// identified from the parse tree
 	ResolveImports(tree ParseTree) ([]*ast.ImportNode, error)
+
+	// ResolvePackageHint returns the package name hint for an imported module
+	//
+	// eg. for a python module "foo.bar" it should return "foo"
+	ResolvePackageHint(moduleName string) (string, error)
```

Comment:
I think `PackageHint` is a concept within the `DependencyUsage` plugin domain. Should we not keep it there? The reason I am thinking like this is a `package` in this context is a `dependency` and this concept related to `DependencyUsage` and probably not a core language concept.

Is there a challenge if we keep the logic of mapping a import name to `packageHint` within the dependency usage plugin only for the different supported languages?

Reply:
agreed, moving this to depsusage